feature/selector.py

```diff
@@ -381,14 +504,19 @@ def __init__(self, selection_method: Union[SelectionMethod.Correlation,
                                 self.selection_method.regularization, self.selection_method.alpha)
         elif isinstance(selection_method, SelectionMethod.TreeBased):
             self._imp = _TreeBased(self.seed, self.selection_method.num_features, self.selection_method.estimator)
+        elif isinstance(selection_method, SelectionMethod.TextBased):
+            self._imp = _TextBased(self.seed, self.selection_method.num_features,
+                                   self.selection_method.featurization_method,
+                                   self.selection_method.optimization_method, self.selection_method.cost_metric,
+                                   self.selection_method.trials)
         elif isinstance(selection_method, SelectionMethod.Statistical):
             self._imp = _Statistical(self.seed, self.selection_method.num_features, self.selection_method.method)
         elif isinstance(selection_method, SelectionMethod.Variance):
             self._imp = _Variance(self.seed, self.selection_method.threshold)
         else:
             raise ValueError("Unknown Selection Method " + str(selection_method))
 
-    def fit(self, data: pd.DataFrame, labels: Optional[pd.Series] = None) -> NoReturn:
+    def fit(self, data: pd.DataFrame, labels: Optional[Union[pd.Series, pd.DataFrame]] = None) -> NoReturn:
```

Comment:
I don't think we need Union inside Optional. Union creates a new type, no? This can be Optional[Series, DF] 

Please also make the change in the fit_transform() functions.

Reply:
This change causes to get error message: "typing.Optional requires a single type. Got (<class 'pandas.core.series.Series'>, <class 'pandas.core.frame.DataFrame'>)."

This is due to the fact that 'typing.Optional' only accepts a single type as argument. 'Optional' required single type, but we passed two types (pd.Series, pd.DataFrame) to fit and fit_transform. So, it is better to use Union to specify that both types are allowed.
I think it is also the reason that @bkleyn gets error for passing the tests. 
Shall we continue with having  `Union'?

Reply:
I see.. Possible that I forgot how Typing works and it seems we "have to" put a Union[x, y] inside the Optional. @bkleyn 

IF this is the case, can you make sure this is consistent across all FIT(), TRANSFORM() calls all the way from public api in selector.py to under the hood in TextBased

Reply:
I think we have to use Union. 

@skadio you are right that it effectively creates a new type. This is what we have in other libs as far as I can see.

Reply:
cool! thanks for raising @zohrehraziei  and thanks for confirming @bkleyn